connect-inject/endpoints_controller.go

```diff
@@ -442,7 +448,86 @@ func (r *EndpointsController) Logger(name types.NamespacedName) logr.Logger {
 func (r *EndpointsController) SetupWithManager(mgr ctrl.Manager) error {
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&corev1.Endpoints{}).
-		Complete(r)
+		Watches(
+			&source.Kind{Type: &corev1.Pod{}},
+			&handler.EnqueueRequestsFromMapFunc{ToRequests: handler.ToRequestsFunc(r.requestsForRunningAgentPods)},
+			builder.WithPredicates(predicate.NewPredicateFuncs(r.filterAgentPods)),
+		).Complete(r)
+}
+
+// filterAgentPods receives meta and object information for Kubernetes resources that are being watched,
+// which in this case are Pods. It only returns true if the Pod is a Consul Agent Pod. It reads the labels
+// from the meta of the resource and uses the values of the "app" and "component" label to validate that
+// the Pod is a Consul Agent.
+func (r EndpointsController) filterAgentPods(meta metav1.Object, object runtime.Object) bool {
+	labels := meta.GetLabels()
+	app, ok := labels["app"]
+	if !ok {
+		return false
+	}
+	component, ok := labels["component"]
+	if !ok {
+		return false
+	}
+	// todo: do we want to filter on other labels?
```

Comment:
I believe we had talked about filtering on `release` as well?
I don't think it is totally required, but the more strict we are the better?

Reply:
That does make sense. We would need the value of the release to be something that is provided to the command struct which we currently do not perform. I know we want to make this change within the endpoints reconciler as well so we should be able to make the change in both places when we have resolved some of those specifics.

Reply:
DONE 😄